google-account-plugin/src/com/google/cloud/tools/intellij/stats/UsageTracker.java

```diff
@@ -33,4 +34,26 @@ void trackEvent(@NotNull String eventCategory,
       @Nullable String eventLabel,
       @Nullable Integer eventValue);
 
+  PartialTrackingEventAction trackEvent(String category);
+
+  interface SendsEvent {
+
+    void send();
+  }
+
+
+  interface PartialTrackingEventAction extends SendsEvent {
+
+    PartialTrackingEventLabel withAction(String value);
+
+    interface PartialTrackingEventLabel extends SendsEvent {
+
+      PartialTrackingEventValue andLabel(String value);
+
+      interface PartialTrackingEventValue extends SendsEvent {
```

Comment:
We actually have a lot of flexibility with what we can pass/provide. The current parameters enforced by `UsageTracker` don't map 1:1 well conceptually anyway.

The backend basically expects the notions of 'event type' and 'event name'. (It's not even certain if these are strictly technical requirements or something we can by-pass.)

After that, there is a notion of 'metadata' in the backend that we can supply in the form of multiple key-value pairs. This is absolutely optional.

There also some uninteresting fields that we can set/pass, but we don't need to be concerned about them.

However, we are not limited just by 'event type', 'event name', and 'metadata'. We can pass whatever (short) information we want to include, by the means of forming a custom (virtual) 'page url' in the way we want: for example, '/virtual/some-string/another-string/and.so.on/'. (This will need some extra handling when we want to extract certain info from the backend, but not a big deal since it will be simple string operations.)

So, the interface design is largely conceptual and up to our discretion.